Sources/InfomaniakCoreDB/Transactionable.swift

```diff
@@ -0,0 +1,66 @@
+/*
+ Infomaniak Core - iOS
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import RealmSwift
+
+/// Something that standardises the DB transactions API across mobile apps.
+///
+/// Still using Realm types, but this "seam" would be a perfect place to abstract DB vendor.
+///
+public protocol Transactionable {
```

Comment:
Reworked Transactionable protocol to hide realm where possible.

added a `fetchObject:forPrimaryKey` func like discussed.